go-controller/pkg/crd/userdefinednetwork/v1/shared.go

```diff
@@ -128,16 +128,44 @@ type Layer2Config struct {
 	// +optional
 	JoinSubnets DualStackCIDRs `json:"joinSubnets,omitempty"`
 
-	// IPAMLifecycle controls IP addresses management lifecycle.
+	// IPAM section contains IPAM-related configuration for the network.
+	// +optional
+	IPAM *IPAMSpec `json:"ipam,omitempty"`
```

Comment:
nit: the spec suffix kind of makes me hope for a status sub-resource.

Maybe we could call this `IPAMConfig` ? 

Totally opinionated, and you can disregard it.

Reply:
good point, +1 easy change, -1 confused dev